src/plugins/kibana_utils/common/of.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Given a promise awaits it and returns a 3-tuple, with the following members:
+ *
+ * - First entry is either the resolved value of the promise or `undefined`.
+ * - Second entry is either the error thrown by promise or `undefined`.
+ * - Third entry is a boolean, truthy if promise was resolved and falsy if rejected.
+ *
+ * @param promise Promise to convert to 3-tuple.
+ */
+export const of = async <T, E = any>(
+  promise: Promise<T>
+): Promise<[T | undefined, E | undefined, boolean]> => {
+  try {
+    return [await promise, undefined, true];
+  } catch (error) {
+    return [, error, false];
```

Comment:
Not a big deal but this should probably follow the same behavior as above:

```suggestion
    return [undefined, error, false];
```

as they are actually a bit different in implementation (although I'm not sure it would make a difference functionally):

![image](https://user-images.githubusercontent.com/1178348/72110283-0924bb80-32f5-11ea-8f63-c1c542d60869.png)

Reply:
Where is this actually used? I don't see any usages of it.

Reply:
It is used in tests, here:

- `bfetch/public/batching/create_streaming_batched_function.test.ts`

It is very useful in tests, to get the error of a promise or make sure you don't have and `UNHANDLED REJECTION` error.